### PR TITLE
handle edge-cases when setting alias keys

### PR DESCRIPTION
**Pull request by bcoe**
In the ongoing effort to make yargs unsinkable, this pull request handles edge-cases when setting aliases. This behavior used to fail:

```js
var argv = require('yargs')
  .alias('x', 'y')
  .alias('y', 'z')
```

As did this:

```js
var argv = require('yargs')
  .alias('z', 'zm')
  .alias('zoom', 'zoop')
  .alias('zm', 'zoom')
```

We're now clever about merging together alias groups as aliases are defined, solving these transitive edge-cases ... definitely an edge case, but I'm a bit OCD about making yargs handle every edge-case I think of!

reviewers: @nexdrew

### CODE REVIEW

File: test/parser.js [hunk 1]
```diff
@@ -399,6 +399,32 @@ describe('parser tests', function () {
...
+      .alias('zm', 'zoom')
+      .argv
+
+    argv.should.have.property('zoom', 55)
+    argv.should.have.property('z', 55)
+    argv.should.have.property('zm', 55)
+    argv.should.have.property('f', 11)
+  })
+
+  it('should merge two lists of aliases if the collide', function () {
```
Comment (nexdrew) [inline test/parser.js]: s/the/they/

File: index.js [hunk 2]
```diff
@@ -166,7 +166,25 @@ function Argv (processArgs, cwd) {
...
+
+      if (aliases) { // x was an alias itself.
+        aliases.push(y)
+      } else { // x is a new alias key.
+        options.alias[x] = (options.alias[x] || []).concat(y)
+      }
+
+      // wait! perhaps we've created two lists of aliases
+      // that reference each other?
+      if ((options.alias[x] || aliases) && options.alias[y]) {
```
Comment (nexdrew) [inline index.js]: Isn't the first condition in this statement technically moot? Based on the 5 lines above (176-180), you know that either `options.alias[x]` or `aliases` must be valid, you just don't know which one.

**Comment by nexdrew** [comment]
@bcoe You're a bit crazy, but I like your style! :+1:

File: index.js (hunk 2 above)
Comment (bcoe) [inline index.js]: you're right, great catch.

**Comment by nexdrew** [comment]
LGTM :shipit:
